### PR TITLE
[thread-cert] refactor case 6.3.1 and 6.3.2 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_5_1_07_MaxChildCount.py [hunk 1]
```diff
@@ -150,6 +167,48 @@ def test(self):
...
+            _pkts.filter_mle_cmd(MLE_PARENT_RESPONSE).must_next()
+            _pkts.filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next()
+
+        for i in range(1, 5):
+            _pkts = router_pkts.copy().filter_wpan_dst64(pv.vars['MED%d' % i])
+            _pkts.filter_mle_cmd(MLE_PARENT_RESPONSE).must_next()
+            _pkts.filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next()
+
+        # Step 2:The DUT MUST properly forward ICMPv6 Echo Requests to all MED children
+        #        The DUT MUST properly forward ICMPv6 Echo Replies to the Leader
```
Comment (jwhui) [inline tests/scripts/thread-cert/Cert_5_1_07_MaxChildCount.py:190]:
```suggestion
        # Step 2: The DUT MUST properly forward ICMPv6 Echo Requests to all MED children
        #         The DUT MUST properly forward ICMPv6 Echo Replies to the Leader
```

**Comment by jwhui** [comment]
@FiveDimensions , please rebase.

**Comment by FiveDimensions** [comment]
> @FiveDimensions , please rebase.

done, please help to review

File: tests/scripts/thread-cert/Cert_6_3_01_OrphanReattach.py [hunk 2]
```diff
@@ -82,6 +88,31 @@ def test(self):
...
+
+        # Step 2: Remove Router from the network
+        # Step 3: The DUT MUST send three MLE Child Update Requests to its parent
+        for i in range(1, 3):
+            _epkts.filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).filter_wpan_dst64(ROUTER).must_next().must_verify(
+                lambda p: {SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, MODE_TLV} <= set(p.mle.tlv.type))
+
+        # Step 5: The DUT MUST perform the attach procedure with the Leader
+        _epkts.filter_mle_cmd(MLE_PARENT_REQUEST).must_next()
+        _epkts.filter_mle_cmd(MLE_CHILD_ID_REQUEST).must_next()
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_3_01_OrphanReattach.py:110]: also verify parent's extended address?

I think we can have a general routine for such commonly used verification:
```suggestion
        # Step 5: The DUT MUST perform the attach procedure with the Leader
        self.verify_attached(pkts, ED, PARENT)
```

Thoughts?

File: tests/scripts/thread-cert/Cert_6_3_02_NetworkDataUpdate.py [hunk 3]
```diff
@@ -96,6 +100,31 @@ def test(self):
...
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        MED = pv.vars['MED']
+        _epkts = pkts.filter_wpan_src64(MED)
+
+        # Step 1: Ensure the topology is formed correctly
+        _epkts.filter_mle_cmd(MLE_CHILD_ID_REQUEST).filter_wpan_dst64(LEADER).must_next()
```
[the remark this answers is not in the document]
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_3_02_NetworkDataUpdate.py:112]: agress

File: tests/scripts/thread-cert/Cert_6_3_01_OrphanReattach.py (hunk 2 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_6_3_01_OrphanReattach.py]: agree

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Pull request merged**